### PR TITLE
Initial iOS 15 + /private/preboot/procursus support

### PR DESCRIPTION
**Pull request by m1stadev**
Updating the base boostrap for initial iOS 15 support, which will requre the entire bootstrap to be inside of /private/preboot/procursus

TODO:
- [x] Fix system-cmds building
- [x] Fix sudo building
- [x] Fix base building
- [x] Disable firmware-sbin building
- [ ] General testing
- [x] Update libiosexec tbd

### CODE REVIEW

**Review by CRKatri — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -124,6 +124,44 @@ ON_DEVICE_SDK_PATH   := $(MEMO_PREFIX)$(MEMO_SUB_PREFIX)/share/SDKs/iPhoneOS.sdk
 BARE_PLATFORM        := iPhoneOS
 export IPHONEOS_DEPLOYMENT_TARGET
 
+else ifeq ($(MEMO_TARGET),iphoneos-arm64-preboot)
+ifneq ($(MEMO_QUIET),1)
+$(warning Building for iOS with preboot prefix)
+endif # ($(MEMO_QUIET),1)
+MEMO_ARCH            := arm64
+PLATFORM             := iphoneos
+DEB_ARCH             := iphoneos-arm
```
Comment (CRKatri) [inline Makefile:133]: Change this to `iphoneos-arm64`

**Review by CRKatri — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (m1stadev) [reply Makefile]: Done.

**Comment by asdfugil** [comment]
there's also watchOS, tvOS, bridgeOS and audioOS.

**Comment by m1stadev** [comment]
That can be added once we know for a fact that SSV actually exists on the root volume on tvOS 15 (probably does, but should be sure, and iirc nobody's confirmed it yet)

**Comment by asdfugil** [comment]
also needs iphoneos-arm migration

**Comment by asdfugil** [comment]
Tweaks *should* work, accroding the situation with rootlessjb.

**Comment by asdfugil** [comment]
- Automatically Move stuff to /p/p/p
  - Move stuff in /L/MS/DL to /p/p/p/lib/TweakInject
- Mach-O linked libraries header patch
  - Append /p/p/p, remove /usr for non-system libraries
  - Inject libiosexec
- Resign
- dpkg should print a warning

**Comment by m1stadev** [comment]
We’re keeping `/usr`, so the path will be `/p/p/p/usr`

**Comment by asdfugil** [comment]
Will we do usrmerge?

**Pull request merged**